waveorder/visuals/matplotlib_visuals.py

```diff
@@ -0,0 +1,296 @@
+import matplotlib.pyplot as plt
+
+from waveorder.visuals.utils import complex_tensor_to_rgb
+from waveorder.sampling import nd_fourier_central_cuboid
+
+import numpy as np
+import os
+import torch
+
+
+def plot_transfer_function(
+    sfZYX_data,
+    filename,
+    zyx_scale,
+    z_slice,
+    s_labels,
+    f_labels,
+    rose_path=None,
+    inches_per_column=1,
+    saturate_clim_fraction=1.0,
+    trim_edges=0,
+):
+    S, F, Z, Y, X = sfZYX_data.shape
+    voxel_scale = [
+        Z * zyx_scale[0],
+        Y * zyx_scale[1],
+        X * zyx_scale[2],
+    ]
+
+    sfZYX_data = nd_fourier_central_cuboid(
+        sfZYX_data, (S, F, Z - trim_edges, Y - trim_edges, X - trim_edges)
+    )
+
+    S, F, Z, Y, X = sfZYX_data.shape
+    sfZYX_data = torch.fft.ifftshift(sfZYX_data, dim=(-3, -2, -1))
+    sfZYX_rgb = complex_tensor_to_rgb(sfZYX_data, saturate_clim_fraction)
+
+    assert S == len(s_labels)
+    assert F == len(f_labels)
+
+    X_size = 1 * inches_per_column
+    Y_size = (zyx_scale[2] / zyx_scale[1]) * inches_per_column
+    Z_size = (zyx_scale[2] / zyx_scale[0]) * inches_per_column
+
+
+    n_rows = S + 1
+    n_cols = (F * 3) + 1
+
+    height = n_rows
+    width = n_cols
+
+    fig, axes = plt.subplots(
+        n_rows,
+        n_cols,
+        figsize=(width, height),
+        gridspec_kw={
+            "wspace": 0.1,  # Adjust this value to reduce space between columns
+            "hspace": 0.1,  # Adjust this value to reduce space between rows
+        },
+    )
+
+    rose_path = os.path.join(
+        os.path.dirname(__file__),
+        f"./assets/rose.png",
+    )
```

Comment:
@talonchandler getting an error due to these lines:
`FileNotFoundError: [Errno 2] No such file or directory: '/hpc/mydata/shalin.mehta/code/waveorder/waveorder/visuals/./assets/rose.png'`

Reply:
Pardon me. Fixed.